chisel/src/main.rs

```diff
@@ -7,360 +7,110 @@ extern crate serde_yaml;
 
 #[macro_use]
 mod logger;
+mod cmd_oneliner;
+mod cmd_run;
 mod config;
 mod driver;
 mod options;
 mod result;
 
-use std::fs::{read, read_to_string, write};
 use std::process;
 
-use clap::{App, Arg, ArgMatches, SubCommand};
-use serde_yaml::Value;
+use clap::{crate_description, crate_name, crate_version, App, Arg, SubCommand};
 
-use libchisel::{
-    checkstartfunc::*, deployer::*, dropsection::*, remapimports::*, remapstart::*, repack::*,
-    snip::*, trimexports::*, trimstartfunc::*, verifyexports::*, verifyimports::*,
-};
+use cmd_oneliner::chisel_oneliner;
+use cmd_run::chisel_run;
+use options::ChiselFlags;
 
-#[cfg(feature = "binaryen")]
-use libchisel::binaryenopt::*;
-
-use libchisel::*;
-
-use config::*;
-
-// Error messages
-static ERR_NO_SUBCOMMAND: &'static str = "No subcommand provided.";
-static ERR_FAILED_OPEN_BINARY: &'static str = "Failed to open wasm binary.";
-static ERR_DESERIALIZE_MODULE: &'static str = "Failed to deserialize the wasm binary.";
-
-// Other constants
-static DEFAULT_CONFIG_PATH: &'static str = "chisel.yml";
-
-/// Chisel configuration structure. Contains a file to chisel and a list of modules configurations.
-struct ChiselContext {
-    ruleset_name: String,
-    file: String,
-    // Output file. If a ModuleTranslator or ModuleCreator is invoked, resorts to a default.
-    outfile: Option<String>,
-    modules: Vec<ModuleContext>,
-}
-
-struct ModuleContext {
-    module_name: String,
-    preset: String,
-}
-
-impl ChiselContext {
-    fn name(&self) -> &String {
-        &self.ruleset_name
-    }
-
-    fn file(&self) -> &String {
-        &self.file
-    }
-
-    fn outfile(&self) -> &Option<String> {
-        &self.outfile
-    }
-
-    fn get_modules(&self) -> &Vec<ModuleContext> {
-        &self.modules
-    }
-}
-
-impl ModuleContext {
-    fn fields(&self) -> (&String, &String) {
-        (&self.module_name, &self.preset)
-    }
+fn fail(code: i32, message: &str) -> ! {
```

Comment:
Looking at all the users of `fail`, all of them use `code = 1`. Do we need the `code` right now?